### PR TITLE
Show more

### LINKED ISSUE #775: Add button show more to product sample
Current [product sample](https://ampbyexample.com/samples_templates/product_page/preview/) shows only 4 related product.
We could add a button "Show more" which will fetch a longer list by using amp-list and changing the source by using amp-bind to show more products, some code [here](https://github.com/ampproject/amphtml/blob/18733f77f2f6562ebffab7cd7891bedf31223e63/test/fixtures/amp-bind-integrations.html#L133).
@ericlindley-g, @choumx  FYI

### PR DESCRIPTION
**Pull request by kul3r4**
Fixes #775 
@sebastianbenz PTAL

### CODE REVIEW

File: src/50_Samples_%26_Templates/Product_Page.html [hunk 1]
```diff
@@ -662,7 +668,7 @@ <h4 class="p2">Products that have been bought together</h4>
         </a>
       </template>
     </amp-list>
-
+    <button class="ampstart-btn caps m1 mb3" id="listSrcButton" on="tap:AMP.setState({product: {relatedProductsSrc: '/json/more_related_products.json'}})">Show more</button>
```
Comment (sebastianbenz) [inline src/50_Samples_%26_Templates/Product_Page.html]: Can you add documentation?

**Comment by kul3r4** [comment]
@sebastianbenz as documented in #775, check disable the web-worker experiment `AMP.toggleExperiment('web-worker')` to solve the error

File: src/50_Samples_%26_Templates/Product_Page.html (hunk 1 above)
Comment (kul3r4) [inline src/50_Samples_%26_Templates/Product_Page.html]: @sebastianbenz Added, thanks!

File: src/50_Samples_%26_Templates/Product_Page.html [hunk 2]
```diff
@@ -647,7 +647,9 @@ <h4 class="p2">Products that have been bought together</h4>
     </div>
     <!--
     With AMP, you can easily pull in different latest offers or highlights without changing the page. To do so, just use `amp-list` to fire a CORS request to a JSON endpoint which supplies the list of related products. These are populated into an amp-mustache template on the client.
-     Learn more about `amp-list` [here](/components/amp-list/)
+    You can dinamically change the source of the `amp-list` CORS JSON endpoint by using `amp-bind`: we are using a variable called `relatedProductsSrc` and we are binding its value
```
Comment (sebastianbenz) [inline src/50_Samples_%26_Templates/Product_Page.html]: s/dinamically/dynamically

Comment (sebastianbenz) [inline src/50_Samples_%26_Templates/Product_Page.html]: remove CORS JSON as it's not relevant here.

Comment (sebastianbenz) [inline src/50_Samples_%26_Templates/Product_Page.html]: suggestion, rephrase to "we bind the variable `relatedProductsSrc` to the to the `src` attribute of the `amp-list`"

File: src/50_Samples_%26_Templates/Product_Page.html [hunk 3]
```diff
@@ -668,6 +670,9 @@ <h4 class="p2">Products that have been bought together</h4>
         </a>
       </template>
     </amp-list>
+    <!--
+      You can set the value of the `relatedProductsSrc` variable by using a button `tap` action.
```
Comment (sebastianbenz) [inline src/50_Samples_%26_Templates/Product_Page.html]: suggestion: "The button's tap action sets the value of the `relatedProductsSrc` to an extended list of related products."

**Comment by kul3r4** [comment]
@sebastianbenz PTAL

**Comment by kul3r4** [comment]
@choumx Do you have any idea why` web-worker` experiment is conflicting with amp-list & ampbind?

**Comment by choumx** [comment]
Yes, working on a fix. Context: https://github.com/ampproject/amp-by-example/issues/775#issuecomment-298730480

**Comment by choumx** [comment]
Should be fixed now. Please try again!

**Comment by kul3r4** [comment]
@choumx Do I need to wait for next canary release? I still get 1493420395795

**Comment by choumx** [comment]
No, it should be working now. The main runtime's version will be the same -- the difference is that another JS file (the web worker) will now serve a different version when `dev-channel` is enabled.

**Comment by kul3r4** [comment]
@choumx I still get the error, I deployed it on [staging](https://amp-by-example-staging.appspot.com/samples_templates/product_page/preview/)

**Comment by choumx** [comment]
Sorry the issue was more involved than I realized (https://github.com/ampproject/amphtml/issues/9154). We had a new prod push today and I don't see the error on your staging link anymore.

**Comment by kul3r4** [comment]
@choumx thanks!

**Pull request merged**